File: C#/PokerWorld/Game/RandomDealer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility.Games.CardGame;

namespace PokerWorld.Game
{
    public class RandomDealer : AbstractDealer
    {
        public override GameCard[] DealHoles(PlayerInfo p)
        {
            GameCard[] set = new GameCard[2];
            set[0] = m_Deck.Pop();
            set[1] = m_Deck.Pop();
            return set;
        }

        public override GameCard[] DealFlop()
        {
            GameCard[] set = new GameCard[3];
            set[0] = m_Deck.Pop();
            set[1] = m_Deck.Pop();
            set[2] = m_Deck.Pop();
            return set;
        }

        public override GameCard DealTurn()
        {
            return m_Deck.Pop();
        }

        public override GameCard DealRiver()
        {
            return m_Deck.Pop();
        }

        public override void FreshDeck()
        {
            m_Deck = GameCardUtility.GetShuffledDeck(false);
        }
    }
}

File: trunk/JAVA/PokerWorld/src/poker/game/dealer/RandomDealer.java
package poker.game.dealer;

import poker.game.PlayerInfo;
import game.Card;
import game.CardSet;

public class RandomDealer extends AbstractDealer
{
    public RandomDealer()
    {
        super();
    }
    
    @Override
    public CardSet dealFlop()
    {
        final CardSet set = new CardSet(3);
        set.add(m_deck.pop());
        set.add(m_deck.pop());
        set.add(m_deck.pop());
        return set;
    }
    
    @Override
    public CardSet dealHoles(PlayerInfo p)
    {
        final CardSet set = new CardSet(2);
        set.add(m_deck.pop());
        set.add(m_deck.pop());
        return set;
    }
    
    @Override
    public Card dealRiver()
    {
        return m_deck.pop();
    }
    
    @Override
    public Card dealTurn()
    {
        return m_deck.pop();
    }
    
    @Override
    public void freshDeck()
    {
        m_deck = CardSet.shuffledDeck();
    }
    
}
